.github/workflows/builder_docker-based_slsa3.yml

```diff
@@ -143,90 +155,118 @@ jobs:
+          retention-days: 5
 
   ###################################################################
   #                                                                 #
   #     Invoke the builder binary to create a buildDefinition       #
   #                                                                 #
   ###################################################################
   generate-build-definition:
-    # TODO: Use the builder binary.
     outputs:
       build-definition-name: ${{ steps.generate.outputs.build-definition-name }}
       build-definition-sha256: ${{ steps.upload.outputs.sha256 }}
     runs-on: ubuntu-latest
-    needs: [detect-env, verify-builder-image]
+    needs: [rng, detect-env, verify-builder-image, generate-builder]
     steps:
+      - name: Checkout builder repository
+        uses: slsa-framework/slsa-github-generator/.github/actions/secure-builder-checkout@main
+        with:
+          repository: "${{ needs.detect-env.outputs.repository }}"
+          ref: "${{ needs.detect-env.outputs.ref }}"
+          path: __BUILDER_CHECKOUT_DIR__
+
+      - name: Download builder
+        uses: ./__BUILDER_CHECKOUT_DIR__/.github/actions/secure-download-artifact
+        with:
+          name: "${{ env.BUILDER_BINARY }}-${{ needs.rng.outputs.value }}"
+          path: "${{ env.BUILDER_BINARY }}"
+          sha256: "${{ needs.generate-builder.outputs.builder-binary-sha256 }}"
+          set-executable: true
+
+      - name: Get source ref
+        id: resolve
+        env:
+          SOURCE_DIGEST: ${{ inputs.source-digest }}
+          GITHUB_SHA: ${{ github.sha }}
+        run: |
+          if [ -z "$SOURCE_DIGEST" ]; then
+            ref=$(echo ${SOURCE_DIGEST} | cut -d':' -f2)
+            echo "ref=$ref" >> $GITHUB_OUTPUT
+          else
+            echo "ref=$GITHUB_SHA" >> $GITHUB_OUTPUT
+          fi
+
+      # TODO: Use secure-checkout, but this requires configuring the source-repository.
+      - name: Checkout the source repository
+        uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+        with:
+          fetch-depth: 1
+          persist-credentials: false
+          path: __PROJECT_CHECKOUT_DIR__
+          repository: ${{ inputs.source-repository || github.repository }}
+          ref: ${{ steps.resolve.outputs.ref }}
+
       - name: Generate build definition
         id: generate
         # These are the inputs, it may be with: for an action or
         # specified with these env vars.
         env:
           BUILDER_IMAGE: ${{ inputs.builder-image }}
           BUILDER_DIGEST: ${{ inputs.builder-digest }}
-          BUILDER_VERIFIED: ${{ needs.verify-builder-image.outputs.verified }}
           CONFIG_PATH: ${{ inputs.config-path }}
-          OUTPUT: ${{ inputs.builder-output-path }}
           # Either the provided source repository or the caller's GITHUB_REPOSITORY
           SOURCE_REPOSITORY: ${{ inputs.source-repository || github.repository }}
```

Comment:
I think it is better to drop `inputs.source-repository`. Unless we want to allow building artefacts from arbitrary repositories. In that case we have to in `--force-checkout` to the build subcommand.

Reply:
done! workflow now performs the checkout too and drops `--force-checkout`